catalyst/data/dataset/self_supervised.py

```diff
@@ -0,0 +1,88 @@
+from typing import Any, Callable, Dict
+
+from torch.utils.data import Dataset
+
+
+class SelfSupervisedDatasetWrapper(Dataset):
+    """The Self Supervised Dataset.
+
+    The class implemets contrastive logic (see Figure 2 from `A Simple Framework
+    for Contrastive Learning of Visual Representations`_.)
+
+    Example:
+
+    .. code-block:: python
+
+        import torchvision
+        from torchvision.datasets import CIFAR10
+
+        from catalyst.contrib.data.datawrappers import SelfSupervisedDatasetWrapper
+
+        transforms = torchvision.transforms.Compose(
+            [
+                torchvision.transforms.RandomResizedCrop(32),
+                torchvision.transforms.ToTensor(),
+                torchvision.transforms.Normalize(
+                    [0.4914, 0.4822, 0.4465], [0.2023, 0.1994, 0.2010]
+                ),
+                torchvision.transforms.ColorJitter(0.8, 0.8, 0.8, 0.2),
+            ]
+        )
+
+        cifar_dataset = CIFAR10(root="./data", download=True, transform=None)
+        cifar_contrastive = SelfSupervisedDatasetWrapper(cifar_dataset, transforms=transforms)
+
+    .. _`A Simple Framework for Contrastive Learning of Visual Representations`:
+        https://arxiv.org/abs/2002.05709
+    """
+
+    def __init__(
+        self,
+        dataset: Dataset,
+        transforms: Callable = None,
+        transform_left: Callable = None,
+        transform_right: Callable = None,
+        transform_original: Callable = None,
+    ) -> None:
+        """
+        Args:
+            dataset: original dataset for augmentation
+            transforms: transforms which will be applied to original batch to get both
+            left and right output batch.
+            transform_left: transform only for left batch
+            transform_right: transform only for right batch
+            transform_original: transforms which will be applied to save original in batch
+
+        """
```

Comment:
could you please move the docs to the class specification also? jupyter help works with them at first

Reply:
Done